### ISSUE TITLE
Take into account earnings after claiming benefits

### ISSUE THREAD

**Issue by Gregable** [Gregable/social-security-tools#54]
Not sure what the formula is for this, but shouldn't be too hard.

**Comment by stkeros** [comment]
It's the same PIA formula. It's just that the additional earnings update the AIME, then then based on the where this falls on the bendpoints, increases the PIA. So while the benefit at age 67 may be $1000, if one works during that year, the benefit at age 68 increases.

One issue this will lead to if there are earnings after the benefit is that the horizontal bar graph won't be straight. There will be jumps or a staircase-like shape.

Note of course that earnings at age 60+ are not indexed. But you are assuming no change in wage growth, so in a way, you are no longer indexing any earnings which are not in the past, regardless of age.

**Comment by Gregable** [comment]
I see. Does this move the spousal benefits up as well for the spouse of the earner with future earnings?

I don't know how much it matters in this tool though. If you've already filed, there isn't much to decide - you already know your benefit (as you are getting the checks) and have made all of your decisions.

So the cases where it matters are:

Suspending filing which is a can of corner cases and complexity i don't want to mess with right now.

The user enters future years predicted earnings and sets the slider to file in the future, but before the last of the future years earnings. And even in this case, the only 'incorrect' bit is that the dates between filing and the final earnings will be a little lower than what the chart shows, since the chart is essentially backdating the earnings to your PIA from the moment you file.

What I might do is detect this specific scenario and display just a line or two of text indicating that the estimates may be incorrect in the years between filing and their final earnings.

**Comment by stkeros** [comment]
Yes, if already receiving a benefit, and worker has earnings after that benefit, the PIA increases and thus the benefit will increase. E.g. if I've been receiving a benefit since age 62 and am not working, it will increase each year only due to the COLA, if any. But if I work at age 64, then sometime during age 65 my benefit will increase due to the increase in PIA when the previous year's earnings are reported.

I agree with you that this is out of scope. I note/asterisk sounds appropriate if the worker's projected earnings overall with claiming. You can just assume that people will have stopped work prior to collecting a benefit. 

Maybe some sort of popup or other notification can be shows as soon as the person attempts to move the slider below the age at which they entered benefits. But what about the case where a 50 year old man selects 35 more years of earnings? Another thing to consider might be to limit the number of years of future earnings based on age, such that one cannot enter earnings past age 69?

**Comment by stkeros** [comment]
Forgot to answer: yes, the spouses benefit also goes up, because the spouses benefit is based on the PIA of the higher earner. If the higher earner's PIA increases, so does the spouse benefit.

**Comment by stkeros** [comment]
Another detail which I think I neglected to mention: if one has earnings while also collecting a benefit, the benefit is actually reduced you collect prior to FRA. That is, there is a penalty of sorts if you collect benefits while working prior to FRA. And the penalty is based on the actual amount of earnings for that year. If you really want to account for this, I guess you have that value in the "earnings slider" set by the user.

This penalty only applies to one's own benefit (spousal or otherwise). For example, if my wife is getting a spousal benefit based on my record, and I am also receiving a benefit and choose to work prior to FRA, MY benefit will be reduced as a result of my earnings, but not my wife's spousal benefits (I'm 99% sure of this, not 100%).

**Comment by stkeros** [comment]
I wonder if one workaround, at first, is to limit the sliders such that earnings cannot occur after age 62. And then consider an update where earnings after age 62 are indeed taken into account.

**Comment by Gregable** [comment]
Similarly, I could limit the slider for start date until after the last earnings date.

However, neither of these solves the spousal case. If the spouse has earnings at age 65, the PIA calculation is made once and then copied into the tool again separately, so the date information has been lost.

**Comment by stkeros** [comment]
One eventual solution may be to have salary info for spouses entered in parallel rather than series. For example, via tabs, one for self and one for spouse, with the third tab for outputs? Rather than PIA-only for spouse. No data loss in that case. And perhaps have a way that one could download a html or other file so that they can re-open their data without re-input, if they want to revisit/review their situation?

**Comment by stkeros** [comment]
> Similarly, I could limit the slider for start date until after the last earnings date.
Or how about a warning/popup if the retirement age slider is moved prior to the last earnings date for self, which reminds people that earnings after the start date aren't taken into account. But you are right, it's a problem for spouse, if the PIA a user entered was calculated with any earnings in the 62-70 years.
